spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/actions/ExpireSnapshotsSparkAction.java

```diff
@@ -79,7 +79,7 @@ public class ExpireSnapshotsSparkAction extends BaseSparkAction<ExpireSnapshotsS
   private Consumer<String> deleteFunc = null;
   private ExecutorService deleteExecutorService = null;
   private Dataset<FileInfo> expiredFileDS = null;
-  private boolean cleanExpiredMetadata = false;
+  private Boolean cleanExpiredMetadata = null;
```

Comment:
@gaborkaszab since the change is tiny, could you please update Spark 3.4 + 3.5 within the same PR please?